File: tests/fixtures/python/global_conditionals.py
"""Module-level conditional logic and deep global nesting."""

import os
import sys

DEBUG = os.environ.get("DEBUG", "false") == "true"

if sys.platform == "win32":
    PATH_SEP = "\\"
    if os.environ.get("CYGWIN"):
        PATH_SEP = "/"
        if os.environ.get("MSYSTEM"):
            PATH_SEP = "/"
else:
    PATH_SEP = "/"

if DEBUG:
    import logging
    logging.basicConfig(level=logging.DEBUG)


def clean_function():
    return 42

File: tests/fixtures/zig/global_conditionals.zig
const std = @import("std");
const builtin = @import("builtin");

var global_state: i32 = 0;

pub fn init() void {
    global_state = 1;
}

// Top-level conditional logic
fn setup() void {
    if (global_state > 0) {
        if (global_state < 100) {
            global_state += 1;
        }
    }
}
